src/rabbit_queue_location_validator.erl

```diff
@@ -33,7 +33,10 @@
 
 validate_policy(KeyList) ->
     case proplists:lookup(<<"queue-master-locator">> , KeyList) of
-        {_, Strategy} -> validate_strategy(Strategy);
+        {_, Strategy} -> case validate_strategy(Strategy) of
```

Comment:
I must be missing something, but I don't see how `validate_strategy/1` can return anything other than `{ok, _}` or `{error, "~p invalid queue-master-locator value", [Strategy]}`

Reply:
`validate_policy/1` should return either `ok` or `{error, term()}`, but not `{ok, term()}`. https://github.com/rabbitmq/rabbitmq-server/blob/master/src/rabbit_policy.erl#L528

Reply:
I see now, thanks.